docs-src/go/chapter-durable-execution/non-deterministic-code-changes.md

```diff
@@ -60,6 +60,10 @@ Let's take a closer look:
 - `WorkflowTaskCompleted`: This Event indicates that the Worker suspended execution and made as much progress that it could.
 - `ActivityTaskScheduled`: This Event indicates that the ExecuteActivity API was called and the Worker sent the [`ScheduleActivityTask`](/references/commands#scheduleactivitytask) Command to the Server.
 - `ActivityTaskStarted`: This Event indicates that the Worker successfully polled the Activity Task and started evaluating Activity code.
+  The ActivityTaskStarted event is written into Workflow history when your activity completes or fails after all of its retries.
+  It may be counter-intuitive that this happens after the terminal Event (like [ActivityTaskCompleted](/references/events#activitytaskcompleted) or [ActivityTaskFailed](/references/events#activitytaskfailed)).
+  Don't be misled into thinking that the activity is failing to start.
+  See [When does Temporal write the ActivityTaskStarted event into Workflow history?](https://community.temporal.io/t/when-does-temporal-write-the-activitytaskstarted-event-into-workflow-history/6162) for more details.
```

Comment:
```suggestion
  See [When does Temporal write the ActivityTaskStarted event into Workflow history?](https://community.temporal.io/t/when-does-temporal-write-the-activitytaskstarted-event-into-workflow-history/6162) for more details.
```

Not a fan of linking to a community forum post where someone has to go to and reduce the ambiguity themselves.
I would prefer it if we can take the essence of that post and make it clear somewhere in our conceptual information.

Reply:
Thanks for your feedback on this. I summarized from the post in the additional lines, passed through Dan D, and offered the link (per the ticket and comments) for additional information in a somewhat constrained bullet point. I tried to balance summary (added) and context (list) and not losing the essential information that was missing from the original.